tests/ballerina-unit-test/src/test/resources/test-src/statements/transaction/transaction_handler_test.bal

```diff
@@ -10,187 +10,242 @@ string idWithinHandler = "";
 function testTransactionStmtWithNoHandlers() returns (string) {
     a = "";
     a = (a + "start");
-    try {
-        transaction {
-            a = a + " inTrx";
-            int i = 0;
-            a = a + " endTrx";
-        } onretry {
-            a = a + " inFailed";
-        }
-    } catch (error err) {
-        a = a + err.message;
+    var result = trap testTransactionStmtWithNoHandlersHelper();
```

Comment:
No need to use trap or helper functions to test this behavior. Use them only transaction and panic behaviors are tested together. Check all the places.